files/grest/cache_queries/epoch_info_cache_setup.sql

```diff
@@ -0,0 +1,204 @@
+    tas.i_active_stake AS i_active_stake,
+    ep.min_fee_a AS p_min_fee_a,
+    ep.min_fee_b AS p_min_fee_b,
+    ep.max_block_size AS p_max_block_size,
+    ep.max_tx_size AS p_max_tx_size,
+    ep.max_bh_size AS p_max_bh_size,
+    ep.key_deposit AS p_key_deposit,
+    ep.pool_deposit AS p_pool_deposit,
+    ep.max_epoch AS p_max_epoch,
+    ep.optimal_pool_count AS p_optimal_pool_count,
+    ep.influence AS p_influence,
+    ep.monetary_expand_rate AS p_monetary_expand_rate,
+    ep.treasury_growth_rate AS p_treasury_growth_rate,
+    ep.decentralisation AS p_decentralisation,
+    encode(ep.entropy, 'hex') AS p_entropy,
+    ep.protocol_major AS p_protocol_major,
+    ep.protocol_minor AS p_protocol_minor,
+    ep.min_utxo_value AS p_min_utxo_value,
+    ep.min_pool_cost AS p_min_pool_cost,
+    encode(ep.nonce, 'hex') AS p_nonce,
+    ep.block_id AS p_block_id,
+    ep.cost_models AS p_cost_models,
+    ep.price_mem AS p_price_mem,
+    ep.price_step AS p_price_step,
+    ep.max_tx_ex_mem AS p_max_tx_ex_mem,
+    ep.max_tx_ex_steps AS p_max_tx_ex_steps,
+    ep.max_block_ex_mem AS p_max_block_ex_mem,
+    ep.max_block_ex_steps AS p_max_block_ex_steps,
+    ep.max_val_size AS p_max_val_size,
+    ep.collateral_percent AS p_collateral_percent,
+    ep.max_collateral_inputs AS p_max_collateral_inputs,
+    ep.coins_per_utxo_word AS p_coins_per_utxo_word
+FROM
+    epoch e
+    INNER JOIN epoch_param ep ON ep.epoch_no = e.no
+    INNER JOIN block b ON b.id = ep.block_id
+    INNER JOIN temp_active_stake tas ON tas.epoch_no = e.no
+    INNER JOIN temp_last_block_time tlbt ON tlbt.epoch_no = e.no
+    INNER JOIN temp_epoch_start_end_time teset ON teset.epoch = e.no;
+
+-- Trigger for updating current epoch data
+DROP FUNCTION IF EXISTS grest.epoch_info_update CASCADE;
+
+CREATE FUNCTION grest.epoch_info_update ()
+    RETURNS TRIGGER
+    AS $epoch_info_update$
+DECLARE
+    _current_epoch integer DEFAULT NULL;
+BEGIN
+    SELECT
+        max(epoch)
+    FROM
+        grest.epoch_info_cache INTO _current_epoch;
+    UPDATE
+        grest.epoch_info_cache
+    SET
+        i_out_sum = update_table.out_sum,
+        i_fees = update_table.fees,
+        i_tx_count = update_table.tx_count,
+        i_blk_count = update_table.blk_count,
+        i_last_block_time = update_table.time
+    FROM (
+        SELECT
+            e.out_sum,
+            e.fees,
+            e.tx_count,
+            e.blk_count,
+            b.time
+        FROM
+            epoch e
+            INNER JOIN block b ON b.block_no = (
+                SELECT
+                    max(block_no)
+                FROM
+                    block)
+            WHERE
+                e.no = _current_epoch) update_table
+WHERE
+    epoch = _current_epoch;
+    RETURN NEW;
+END;
+$epoch_info_update$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS epoch_info_update_trigger ON public.block;
+
+CREATE TRIGGER epoch_info_update_trigger
+    AFTER INSERT ON public.block
+    FOR EACH ROW
```

Comment:
Discussed with damjan that we'd prolly want to reduce frequency of this call using intervals, as most (not all) of the information will likely be almost same throughout the epoch, possibly even liaise with cache query intervals, can use 15m for initial mock